src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -106,6 +107,17 @@ public void rebuildSessionFromFrame(MenuSession menuSession, CaseSearchHelper ca
                         }
                     }
                 }
+                if (currentStep == null && processedStepsCount != steps.size()) {
+                    StringJoiner optionsIDJoiner = new StringJoiner(", ", "[", "]");
```

Comment:
I cannot believe `StringJoiner` exists since 1.8 and I never heard of it. Pretty handy.